src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HPack/HPackEncoder.cs

```diff
@@ -291,6 +291,7 @@ public static bool EncodeStringLiterals(string[] values, string separator, Span<
 
                 valueLength = checked((int)(valueLength + (values.Length - 1) * separator.Length));
 
+                destination[0] = 0;
```

Comment:
This is the right fix for 3.0.  But if every call to Encode requires the first byte to be set, should that byte be an argument to Encode, which would then always set it?

Reply:
Possibly. I suspect we will need to rework many aspects of HPack for 5.0